File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution {
  public:
    void sort012(vector<int>& arr) {
        // code here
        // 09.09.24 (potd)
        int lo=0, mid=0, hi=arr.size()-1;
        
        while(mid <= hi){
            if(arr[mid] == 0){
                swap(arr[lo], arr[mid]);
                mid++;
                lo++;
            } else if(arr[mid] == 1){
                mid++;
            } else {
                swap(arr[hi], arr[mid]);
                hi--;
            }
        }
    }
};

//{ Driver Code Starts.
int main() {

    int t;
    cin >> t;
    cin.ignore();
    while (t--) {

        vector<int> a;
        string input;
        int num;

        getline(cin, input);
        stringstream s2(input);
        while (s2 >> num) {
            a.push_back(num);
        }
        Solution ob;
        ob.sort012(a);

        int n = a.size();
        for (int i = 0; i < n; i++) {
            cout << a[i] << " ";
        }

        cout << endl;
    }
    return 0;
}

// } Driver Code Ends

File: Difficulty: Easy/Sort 0s, 1s and 2s/sort-0s-1s-and-2s.py
#{ 
 # Driver Code Starts

# } Driver Code Ends
class Solution:
    # Function to sort an array of 0s, 1s, and 2s
    def sort012(self, arr):
        # code here
        # 05.12.24 potd
        lo, mid, hi = 0, 0, len(arr)-1
        
        while mid <= hi:
            if arr[mid] == 0:
                arr[lo], arr[mid] = arr[mid], arr[lo]
                lo += 1
                mid += 1
            elif arr[mid] == 1:
                mid += 1
            else:
                arr[hi], arr[mid] = arr[mid], arr[hi]
                hi -= 1
        

#{ 
 # Driver Code Starts.
def main():
    t = int(input().strip())  # Read the number of test cases
    ob = Solution()

    while t > 0:
        t -= 1
        arr = list(map(int,
                       input().strip().split())
                   )  # Read the array as space-separated integers
        ob.sort012(arr)  # Sort the array

        print(' '.join(map(str, arr)))  # Print the sorted array
        print("~")
        
if __name__ == "__main__":
    main()

# } Driver Code Ends